### PR TITLE
Removes old AudioPlayers when hot restarted

### PR DESCRIPTION
**Pull request by Afsar-Pasha**
fixes flame-engine/flame/issues/141, tested only in android.

### CODE REVIEW

**Comment by luanpotter** [comment]
Thanks for this! I've been trying to fix this for ages!!!

**Comment by luanpotter** [comment]
@Afsar-Pasha can you please add an entry to the changelog, under [next], with your fix? something like `fixing missing cleanup on hot restart` should suffice. Ideally @erickzanardo can help us testing on ios as well

**Comment by erickzanardo** [comment]
@luanpotter tested on iOS, I get these exceptions when hot reloading:

```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: Reading static variable '_channel@588020947' during its initialization
#0      AudioPlayer._channel (package:audioplayers/audioplayers.dart:123:30)
#1      AudioPlayer._invokeMethod (package:audioplayers/audioplayers.dart:302:12)
#2      new AudioPlayer (package:audioplayers/audioplayers.dart:287:5)
#3      AudioPlayer._doHandlePlatformCall (package:audioplayers/audioplayers.dart:511:25)
<asynchronous suspension>
#4      AudioPlayer.platformCallHandler (package:audioplayers/audioplayers.dart:497:7)
<asynchronous suspension>
#5      MethodChannel._handleAsMethodCall (package:flutter/src/services/platform_channel.dart:402:55)
<asynchronous suspension>
#6      MethodChannel.setMethodCallHandler.<anonymous closure> (package:flutter/src/services/platform_channel.dart:370:54)
#7      _DefaultBinaryMessenger.handlePlatformMessage (package:flutter/src/services/binding.dart:200:33)
<asynchronous suspension>
```

**Comment by luanpotter** [comment]
So maybe we can gate this behind an `is android` if statement? is there a quick way to make it work on ios?

**Comment by erickzanardo** [comment]
The gate could be a good option, I don’t think I will have time to take a better look on this and try to come up with a solution for iOS

File: lib/audioplayers.dart [hunk 1]
```diff
@@ -513,6 +513,15 @@ class AudioPlayer {
 
     final playerId = callArgs['playerId'] as String;
     final AudioPlayer player = players[playerId];
+
+    if (!kReleaseMode && Platform.isAndroid && player == null) {
+      final oldPlayer = AudioPlayer(playerId: playerId);
+      await oldPlayer.release();
+      oldPlayer.dispose();
+      players.remove(playerId);
```
Comment (Afsar-Pasha) [inline lib/audioplayers.dart]: This is broken on `Flutter 1.12.13+hotfix.5 • channel stable`, produces exception `CyclicInitializationError (Reading static variable '_channel@352020947' during its initialization)` 
changing it to
`_channel.invokeMethod('release', {'playerId': playerId});`
works for me but I think it won't work on previous versions of flutter.

**Pull request merged**